streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -341,6 +342,24 @@ public Subscription subscription(final Set<String> topics) {
             clientMetadata.addConsumer(consumerId, info);
         }
 
+        if (futureMetadataVersion != -1) {
+            if (minUserMetadataVersion >= 3) {
+                log.info("Received a future (version probing) subscription (version: {}). Sending empty assignment back (with supported version {}).",
+                    futureMetadataVersion,
+                    SubscriptionInfo.LATEST_SUPPORTED_VERSION);
+                return emptyAssignment(subscriptions);
```

Comment:
Should we still do `taskManager.setClusterMetadata(fullMetadata);` before returning? I'm not sure if it will give us any good but just bringing this up..

Reply:
I don't think it would have any advantage -- with the empty assignment going back, `taskManager` will not be used but only a new rebalance will happen.